machineconfiguration/v1/types.go

```diff
@@ -0,0 +1,348 @@
+package v1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	runtime "k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/util/intstr"
+)
+
+// MachineConfigRoleLabelKey is metadata key in the MachineConfig. Specifies the node role that config should be applied to.
+// For example: `master` or `worker`
+const MachineConfigRoleLabelKey = "machineconfiguration.openshift.io/role"
+
+// +genclient
+// +genclient:noStatus
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// MachineConfig defines the configuration for a machine
+type MachineConfig struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty" protobuf:"bytes,1,opt,name=metadata"`
+
+	Spec MachineConfigSpec `json:"spec" protobuf:"bytes,2,opt,name=spec"`
+}
+
+// MachineConfigSpec is the spec for MachineConfig
+type MachineConfigSpec struct {
+	// OSImageURL specifies the remote location that will be used to
+	// fetch the OS.
+	OSImageURL string `json:"osImageURL" protobuf:"bytes,1,opt,name=osImageURL"`
+	// Config is a Ignition Config object.
+	Config runtime.RawExtension `json:"config" protobuf:"bytes,2,opt,name=config"`
```

Comment:
This is fine.  You're interpretting the value elsewhere but this eliminated the problematic compile time dep?

Reply:
yes. This is parsed into Ignition.Config by the MCO